Sources/VaporToolbox/Util/PrintUtilities.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+
+extension String {
+    fileprivate var centered: String {
+        // Split the string into lines
+        var lines = self.split(separator: Character("\n")).map(String.init)
+
+        guard lines.count > 0 else {
+            return ""
+        }
+
+        // Remove ANSI color codes to get the true length of the string
+        let uncoloredLines = lines.map { $0.removingANSIColors }
+
+        var longestLine = 0
+        for line in uncoloredLines {
+            if line.count > longestLine {
+                longestLine = line.count
+            }
+        }
+
+        // Calculate the padding and make sure it's greater than or equal to 0
+        let padding = max(0, (terminalSize.width - longestLine) / 2)
+
+        // Apply the padding to each line
+        for i in 0..<lines.count {
+            for _ in 0..<padding {
+                lines[i].insert(" ", at: startIndex)
+            }
+        }
```

Comment:
```suggestion
```suggestion
        for i in lines.indices {
            lines[i].insert(String(repeating: " ", count: padding), at: lines[i].startIndex)
        }
```
This was functionally wrong before - `startIndex` unqualified was resolved to `self.startIndex`, which is not necessarily a valid index for each individual line.